readme.md

```diff
@@ -84,6 +84,91 @@ Please see the following reference implementations for additional examples and t
+Initial implementation for BCHN intends to use [The GNU Multiple Precision Arithmetic Library](https://gmplib.org/) (AKA `gmp`), which is a time-tested and performant library, of which we only need a subset of operations for BCH Script arithmetic opcodes.
+The same library was used by libsecp256k1, until 2015 when it was [fully removed](https://github.com/bitcoin/bitcoin/commit/75a880390191aeaf7d2fa326f194349a891db022#diff-242b05bb9c57125ed0fbecbda608e3fa6fd3d7cae41e1cba0dcfd706252a7fc0R413) as a dependency.
+
+There exist big integer libraries for languages other than C++ and used in Bitcoin Cash node implementations:
+
+- Golang's [`math/big`](https://pkg.go.dev/math/big) package, expected to be used with [BCHD](https://github.com/OPReturnCode/bchd), considering it is already internally used for [cryptography operations](https://github.com/OPReturnCode/bchd/blob/67d41ee8838026571863491e88288d6d2d5cb717/bchec/privkey.go#L11).
+- Java's [`java.math,BigInteger`](https://docs.oracle.com/en/java/javase/21/docs/api/java.base/java/math/BigInteger.html) library, expected to be used with [Bitcoin Verde](https://github.com/SoftwareVerde/bitcoin-verde), considering it is already internally used for [uint256 operations](https://github.com/SoftwareVerde/bitcoin-verde/blob/e9d140cac8b93a9db572bda906db9decfac1d7ae/src/main/java/com/softwareverde/bitcoin/block/header/difficulty/work/ChainWork.java#L8). It doesn't need to use it for cryptography operatins, because native secp256k1 implementation exists for Java and [is used by Bitcoin Verde](https://github.com/SoftwareVerde/bitcoin-verde/blob/e9d140cac8b93a9db572bda906db9decfac1d7ae/src/main/java/org/bitcoin/NativeSecp256k1.java#L5).
+
+Similarly, there exist big integer libraries for languages in which Bitcoin Cash smart contract libraries are written:
+
+- JavaScript's [`BigInt`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) library, expected to be used with Bitcoin Cash JavaScript libraries such as [libauth](https://github.com/bitauth/libauth) and [CashScript](https://github.com/CashScript/cashscript).
+- Python [natively supports unlimited integer precision](https://docs.python.org/3/library/stdtypes.html#numeric-types-int-float-complex), expected to be used with [python-bitcoincash](https://github.com/dagurval/python-bitcoincash).
+
+All these implementations must produce exact same results for exact same inputs, else node implementations would risk a "hard" network split and smart contract libraries would risk faulty transaction validation breaking user-facing applications.
+
+This risk is mitigated by having a [comprehensive set of test vectors](#test-vectors), which cover all the arithmetic opcodes and around all the edges, and that ensures low implemnatiton risk as any implementation errors would be caught by the test suite, and before deployment.
```

Comment:
```suggestion
This risk is mitigated by having a [comprehensive set of test vectors](#test-vectors), which cover all the arithmetic opcodes and around all the edges, and that ensures low implementation risk as any implementation errors would be caught by the test suite, and before deployment.
```

Reply:
commited the suggestion for now, but once property tests get merged I think it'd be better to link to tests section rather than implementation section: https://github.com/bitjson/bch-bigint/pull/7/files#diff-5a831ea67cf5cf8703b0de46901ab25bd191f56b320053be9332d9a3b0d01d15R77